### ISSUE TITLE
Undistinct Filenames for Variants in Histoire

### ISSUE THREAD

**Issue by hinogi** [lost-pixel/lost-pixel#366]
https://github.com/lost-pixel/lost-pixel/blob/d00668eabe7a93f4734069929c1a9c862ddfe16c/src/crawler/histoireScreenshots.ts#L38

I guess the fallback at this point should be `${story.id}_${variant.title}`, because you have a `default` variant in almost all stories and that will overwrite the screenshot every time.

**Comment by hinogi** [comment]
@akryum what would you say?

**Comment by hinogi** [comment]
or use the storyid as foldername

**Comment by d-ivashchuk** [comment]
We are aiming to harmonize the features across different sources so it might get better with next releases in the future anyways. 

Overall if the issue is pressing, happy to welcome the contribution and release a new version for you asap

**Comment by hinogi** [comment]
It is kinda pressing because right now, if you have different 5 stories with overlapping variant names last one wins so you only have 1 variant of the last story because all the other variants get overwritten each time a new story is crawled.

**Comment by Akryum** [comment]
Maybe `${story.title ?? 'untitled'}_${variant.title ?? 'default'}` ?

**Comment by hinogi** [comment]
> Maybe `${story.title ?? 'untitled'}_${variant.title ?? 'default'}` ?

is the story title unique?

**Comment by Akryum** [comment]
The issue is probably also related to the id of the screenshot.

https://github.com/lost-pixel/lost-pixel/blob/d00668eabe7a93f4734069929c1a9c862ddfe16c/src/crawler/histoireScreenshots.ts#L44

Should probably be `${story.id}_${variant.id}`.

@hinogi I'm not sure that the screenshot name is used to differentiate between them?

**Comment by hinogi** [comment]
the `shotName` is the resulting filename.
currently
```json
{
  "stories": [
    {
      "id": "story-1",
      "variants": [
        {
          "id": "story-1-0",
          "title": "default"
        }
      ]
    },
    {
      "id": "story-2",
      "variants": [
        {
          "id": "story-2-0",
          "title": "default"
        }
      ]
    }
  ]      
}
```
will only generate a single default.png

**Comment by hinogi** [comment]
> The issue is probably also related to the id of the screenshot.
> 
> https://github.com/lost-pixel/lost-pixel/blob/d00668eabe7a93f4734069929c1a9c862ddfe16c/src/crawler/histoireScreenshots.ts#L44
> 
> Should probably be `${story.id}_${variant.id}${label}`.
> 
> @hinogi I'm not sure that the screenshot name is used to differentiate between them?

`variant.id` would also work I guess because the `story.id` is in there unless altered. But the title is not unique for variants that was the point.

**Comment by d-ivashchuk** [comment]
@hinogi @Akryum i trust your judgement here for the uniqueness. When you are satisfied with the PR, ping me, I will review it shortly and merge!

**Comment by hinogi** [comment]
@d-ivashchuk go go go :D

**Comment by d-ivashchuk** [comment]
@hinogi I've released `3.12.1` right now, it has some other changes apart from the one that you did for histoire, but I hope there is nothing breaking there and you will have the story ids in a correct way 🙌 tell me if there is anything you will need here
